flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -673,4 +682,87 @@ public void testBlockingConnectionsBetweenChainsEnabled() {
 		assertEquals(ResultPartitionType.PIPELINED_BOUNDED, filterAndMapVertex.getProducedDataSets().get(0).getResultType());
 		assertEquals(ResultPartitionType.PIPELINED_BOUNDED, printVertex.getInputs().get(0).getSource().getResultType());
 	}
+
+	@Test
+	public void testOperatorNotSupportedByCheckpointing() {
+
+		// test the operator which implements InputSelectable
+		try {
+			StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+			env.enableCheckpointing(60_000L);
+
+			DataStreamSource<String> source1 = env.fromElements("1");
+			DataStreamSource<Integer> source2 = env.fromElements(1);
+			source1.connect(source2)
+				.transform("test",
+					BasicTypeInfo.STRING_TYPE_INFO,
+					new TestAnyModeReadingStreamOperator("test operator"))
+				.print();
+
+			StreamingJobGraphGenerator.createJobGraph(env.getStreamGraph());
```

Comment:
What about refactoring this test, by extracting building a sample job graph (with operator under the test as a parameter) via a helper method and also, splitting the test into three separate tests?

If they run separately, you can also use simpler structure
```
@Test(expected = UnsupportedOperationException.class)
public void testValidationForOperator...() throws Exception {
    // setup graph:
    ...
    StreamingJobGraphGenerator.createJobGraph(env.getStreamGraph());
    fail("unreachable");
}